### PR TITLE
add UI for comparing arbitrary builds

### PR DESCRIPTION
**Pull request by terceiro**
Every build that has a baseline build now has a "(compare)" link which will lead to the comparison page. You can also compare builds by selecting them in the build list, or by browsing directly to the  compare builds page (Builds - compare builds) and entering build numbers number there.

### CODE REVIEW

**Comment by terceiro** [comment]
just pushed a new commit that adds controls to select builds for comparison
from the build list page

**Comment by terceiro** [comment]
I now consider this ready. I have  update title and description accordingly.

File: frontend/static/templates/build_list.html [hunk 1]
```diff
@@ -9,35 +9,42 @@
...
           <a ng-href="#/build/{{ build.id }}">{{ build.name }}</a> <br/>
           <a style="color:#999" href="{{ build.url }}">{{ build.url }}</a>
         </td>
 
         <td>
-          <div ng-repeat="test in build.test_jobs">
-            (<a href="{{ test.url }}" target="_blank">#{{ test.id }}</a>
-            <ng-include src="'/static/templates/_test_job_status.html'"></ng-include>)
-      </div>
+          {{build.test_jobs|count_by_status}}
```
Comment (mwasilew) [inline frontend/static/templates/build_list.html]: please add new line after each status line.

File: frontend/static/templates/build_list.html [hunk 2]
```diff
@@ -23,6 +26,12 @@
     <tbody>
       <tr ng-repeat="build in page.results">
 
+        <td><input type='radio' name='compareFrom' value='{{build.id}}' ng-click='setCompareFrom(build.id)'/></td>
+        <td ng-show='compareFrom'>
+          <input ng-show='compareFrom != build.id' type='radio' name='compareTo' value='{{build.id}}' ng-click='setCompareTo(build.id)'/>
+          <a ng-show='compareFrom != build.id && compareTo == build.id' href='#/builds/compare/?from={{compareFrom}}&to={{build.id}}' title='compare selected builds'>compare</a>
```
Comment (mwasilew) [inline frontend/static/templates/build_list.html]: this link would look better above/below the table IMHO.

**Comment by mwasilew** [comment]
Works OK. I'm missing some information in case there are no results to compare. This happens in android-qa-reports (not art-reports). I'm only getting empty comparison page. It would be nice to display some information banner like "sth went wrong, no results to display".

File: frontend/static/templates/build_list.html (hunk 1 above)
Comment (terceiro) [inline frontend/static/templates/build_list.html]: done

**Comment by terceiro** [comment]
I think I addressed all the comments now

**Comment by mwasilew** [comment]
Yep, looks good to me now :)

**Pull request merged**